precompile/contract/interfaces.go

```diff
@@ -19,6 +20,29 @@ type StatefulPrecompiledContract interface {
 	Run(accessibleState AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error)
 }
 
+// PredicateContext provides context to stateful precompile predicates
+type PredicateContext struct {
+	SnowCtx            *snow.Context
+	ProposerVMBlockCtx *block.Context
+}
+
+// Predicater is an optional interface for StatefulPrecompiledContracts to implement.
+// If implemented, the predicate will be enforced on every transaction in a block, prior to the block's execution.
+// If VerifyPredicate returns an error, the block will fail verification with no further processing.
+// WARNING: this is not intended to be used for custom precompiles. Backwards compatibility with custom precompiles that
+// use the Predicater interface will not be supported.
+type Predicater interface {
+	VerifyPredicate(predicateContext *PredicateContext, storageSlots []byte) error
+}
+
+// Accepter is an optional interface for StatefulPrecompiledContracts to implement.
+// If implemented, Accept will be called for every log with the address of the precompile when the block is accepted.
+// WARNING: this is not intended to be used for custom precompiles. Backwards compatibility with custom precompiles that
+// use the Accepter interface will not be supported.
+type Accepter interface {
+	Accept(txHash common.Hash, logIndex int, topics []common.Hash, logData []byte) error
```

Comment:
Are we still planning on passing in a unified Log struct? This was originally called out [here](https://github.com/ava-labs/subnet-evm/pull/453/files#r1098366475).

Reply:
No, the refactor will not make this possible.